internal/config/azure.go

```diff
@@ -0,0 +1,228 @@
+	}
+}
+
+// GetVariant returns azure-sev-snp as the variant.
+func (AzureSEVSNP) GetVariant() variant.Variant {
+	return variant.AzureSEVSNP{}
+}
+
+// GetMeasurements returns the measurements used for attestation.
+func (c AzureSEVSNP) GetMeasurements() measurements.M {
+	return c.Measurements
+}
+
+// SetMeasurements updates a config's measurements using the given measurements.
+func (c *AzureSEVSNP) SetMeasurements(m measurements.M) {
+	c.Measurements = m
+}
+
+// EqualTo returns true if the config is equal to the given config.
+func (c AzureSEVSNP) EqualTo(old AttestationCfg) (bool, error) {
+	otherCfg, ok := old.(*AzureSEVSNP)
+	if !ok {
+		return false, fmt.Errorf("cannot compare %T with %T", c, old)
+	}
+
+	firmwareSignerCfgEqual := c.FirmwareSignerConfig.EqualTo(otherCfg.FirmwareSignerConfig)
+	measurementsEqual := c.Measurements.EqualTo(otherCfg.Measurements)
+	bootloaderEqual := c.BootloaderVersion == otherCfg.BootloaderVersion
+	teeEqual := c.TEEVersion == otherCfg.TEEVersion
+	snpEqual := c.SNPVersion == otherCfg.SNPVersion
+	microcodeEqual := c.MicrocodeVersion == otherCfg.MicrocodeVersion
+	rootKeyEqual := bytes.Equal(c.AMDRootKey.Raw, otherCfg.AMDRootKey.Raw)
+
+	return firmwareSignerCfgEqual && measurementsEqual && bootloaderEqual && teeEqual && snpEqual && microcodeEqual && rootKeyEqual, nil
+}
+
+// UnmarshalYAML implements a custom unmarshaler to support setting "latest" as version.
+func (c *AzureSEVSNP) UnmarshalYAML(unmarshal func(any) error) error {
+	aux := &fusedAzureSEVSNP{
+		auxAzureSEVSNP: (*auxAzureSEVSNP)(c),
+	}
+	if err := unmarshal(aux); err != nil {
+		return fmt.Errorf("failed to unmarshal AzureSEVSNP: %w", err)
+	}
+	c = (*AzureSEVSNP)(aux.auxAzureSEVSNP)
+
+	for _, versionType := range []version.Type{version.Bootloader, version.TEE, version.SNP, version.Microcode} {
+		if !convertLatestToNumber(c, versionType, aux) {
+			if err := convertStringToUint(c, versionType, aux); err != nil {
+				return fmt.Errorf("failed to convert %s version to number: %w", versionType, err)
```

Comment:
```suggestion
		return fmt.Errorf("unmarshal AzureSEVSNP: %w", err)
	}
	c = (*AzureSEVSNP)(aux.auxAzureSEVSNP)

	for _, versionType := range []version.Type{version.Bootloader, version.TEE, version.SNP, version.Microcode} {
		if !convertLatestToNumber(c, versionType, aux) {
			if err := convertStringToUint(c, versionType, aux); err != nil {
				return fmt.Errorf("convert %s version to number: %w", versionType, err)
```
avoid phrases like "failed to", "error when", etc. in error strings because it gets too verbose when they get chained